src/main/java/com/jcabi/xml/StrictXML.java

```diff
@@ -192,14 +193,32 @@ private static String join(final Iterable<?> iterable, final String sep) {
     private static Collection<SAXParseException> validate(final XML xml) {
         final Collection<SAXParseException> errors =
             new CopyOnWriteArrayList<SAXParseException>();
+        final int amountOfRetries = 3;
         try {
-            final Schema schema = SchemaFactory
-                .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI).newSchema();
-            final Validator validator = schema.newValidator();
+            final Validator validator = newValidator();
             validator.setErrorHandler(
                 new XSDDocument.ValidationHandler(errors)
             );
-            validator.validate(new DOMSource(xml.node()));
+            final DOMSource domsrc = new DOMSource(xml.node());
+            for (int retry = 0; retry < amountOfRetries; ++retry) {
+                try {
+                    validator.validate(domsrc);
+                    // @checkstyle ModifiedControlVariableCheck (1 line)
+                    retry = amountOfRetries;
+                } catch (final SocketException ex) {
+                    Logger.error(
+                        StrictXML.class,
+                        "Try #%d of %d failed: %s: %s",
+                        retry,
+                        amountOfRetries,
+                        ex.getClass().getName(),
+                        ex.getMessage()
+                    );
+                    if (amountOfRetries == retry + 1) {
```

Comment:
`retry > max` seems impossible within this loop, but it can be changed in a different way, something like `for (int retry = 1; retry <= max; ++retry) {` and `if (retry == max)`